leaf-ui/js/onFunctions.js

```diff
@@ -891,22 +891,32 @@ paper.on({
                     $('.inspector').append(actorInspector.el);
                     actorInspector.render();
                 } else {
-                    elementInspector.render(cell);
+                    var elementInspector = new ElementInspector({model: cell});
+                    $('.inspector').append(elementInspector.el);
+                    elementInspector.render();
                     // if user was dragging element
                     if (evt.data.move) {
                         // unembed intention from old actor
                         if (cell.get('parent')) {
                             graph.getCell(cell.get('parent')).unembed(cell);
-
+                            /** 
                             // remove nodeID from actor intentionIDs list
                             var userIntention = model.getIntentionByID(cell.attributes.nodeID);
                             if (userIntention.nodeActorID !== '-') {
                                 var actor = model.getActorByID(userIntention.nodeActorID);
                                 actor.removeIntentionID(userIntention.nodeID);
                             }
+                            */
                         }
                         // embed element in new actor
-                        embedBasicActor(cell);
+                        var overlapCells = paper.findViewsFromPoint(cell.getBBox().center());
```

Comment:
Can you reset the IntentionBBM actorID variable to '-' if overlapCells.length = 0

Reply:
You don't need it tho. cell.get('parent') automatically gets the node the cell is embedded to

Reply:
If it is unembedded the function doesn't return the parent

Reply:
Oh awesome - should we remove the ActorId param from intentions then?

Reply:
I think so